Upstream: no
Reason:
    on non-glibc, fortify functionality needs to be provided by sys-libs/fortify-headers;
    this makes it included by default so that it can be activated the usual way, by just
    passing -D_FORTIFY_SOURCE=[0|1|2].

--- gcc-6.1.0/gcc/config/linux.h.orig	2016-03-09 13:27:13.289736710 +0000
+++ gcc-6.1.0/gcc/config/linux.h	2016-03-09 13:29:32.295625046 +0000
@@ -146,6 +146,8 @@
 
 #ifdef NATIVE_SYSTEM_HEADER_DIR
 #define INCLUDE_DEFAULTS_MUSL_NATIVE			\
+    { NATIVE_SYSTEM_HEADER_DIR "/fortify", 0, 0, 0, 1, 2 },	\
+    { NATIVE_SYSTEM_HEADER_DIR "/fortify", 0, 0, 0, 1, 0 },	\
     { NATIVE_SYSTEM_HEADER_DIR, 0, 0, 0, 1, 2 },	\
     { NATIVE_SYSTEM_HEADER_DIR, 0, 0, 0, 1, 0 },
 #else
